go/testimonyd/internal/socket/socket.go

```diff
@@ -161,7 +161,11 @@ func (s *socket) reportStats() {
 // run handles new connections, old connections, new blocks... basically
 // everything.
 func (s *socket) run() {
-	go s.getNewBlocks()
+	go func() {
+		if err := s.getNewBlocks(); err != nil { // error handling block
+			log.Print(err)
```

Comment:
I think we'll want to plumb this error up so that run() returns an error if this happens.  Possibly something like:

selectErr := make(chan error, 1)
go func() {
  selectErr <- s.getNewBlock()
}()

then, in the 170-190 select block, add a:

  case err := <-selectErr:
    return fmt.Errorf("getting new block: %v", err)

Alternatively, log.Fatal here, since getNewBlock failing seems catastrophic.

Reply:
@gconnell Moved error to channel and called log.Fatal.